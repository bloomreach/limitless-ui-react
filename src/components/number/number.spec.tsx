import { render } from '@testing-library/react';
import { describe, it, expect } from 'vitest';

import { Number } from './';

describe('Number', () => {
  it('should render successfully for different values and locales', () => {
    const locales = ['en-US', 'nl-NL', 'sk-SK', 'ja-JP', 'hi-IN', 'ar-AE', 'zh-CN'];
    const values = [0.01, 0.1, 1, 10, 100, 1000, 10000, 100000, 1000000];
    const { container: { firstElementChild } } = render(
      <div>
        {values.map(value => {
          return locales.map(locale => {
            return (<Number key={`${value}-${locale}}`} value={value} locale={locale} />);
          })
        })}
      </div>
    );

    expect(firstElementChild).toMatchSnapshot();
  });
});
