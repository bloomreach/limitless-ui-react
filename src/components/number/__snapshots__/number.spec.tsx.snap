// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Number > should render successfully for different values and locales 1`] = `
<div>
  <span
    class="lui-number"
  >
    0.01
  </span>
  <span
    class="lui-number"
  >
    0,01
  </span>
  <span
    class="lui-number"
  >
    0,01
  </span>
  <span
    class="lui-number"
  >
    0.01
  </span>
  <span
    class="lui-number"
  >
    0.01
  </span>
  <span
    class="lui-number"
  >
    0.01
  </span>
  <span
    class="lui-number"
  >
    0.01
  </span>
  <span
    class="lui-number"
  >
    0.1
  </span>
  <span
    class="lui-number"
  >
    0,1
  </span>
  <span
    class="lui-number"
  >
    0,1
  </span>
  <span
    class="lui-number"
  >
    0.1
  </span>
  <span
    class="lui-number"
  >
    0.1
  </span>
  <span
    class="lui-number"
  >
    0.1
  </span>
  <span
    class="lui-number"
  >
    0.1
  </span>
  <span
    class="lui-number"
  >
    1
  </span>
  <span
    class="lui-number"
  >
    1
  </span>
  <span
    class="lui-number"
  >
    1
  </span>
  <span
    class="lui-number"
  >
    1
  </span>
  <span
    class="lui-number"
  >
    1
  </span>
  <span
    class="lui-number"
  >
    1
  </span>
  <span
    class="lui-number"
  >
    1
  </span>
  <span
    class="lui-number"
  >
    10
  </span>
  <span
    class="lui-number"
  >
    10
  </span>
  <span
    class="lui-number"
  >
    10
  </span>
  <span
    class="lui-number"
  >
    10
  </span>
  <span
    class="lui-number"
  >
    10
  </span>
  <span
    class="lui-number"
  >
    10
  </span>
  <span
    class="lui-number"
  >
    10
  </span>
  <span
    class="lui-number"
  >
    100
  </span>
  <span
    class="lui-number"
  >
    100
  </span>
  <span
    class="lui-number"
  >
    100
  </span>
  <span
    class="lui-number"
  >
    100
  </span>
  <span
    class="lui-number"
  >
    100
  </span>
  <span
    class="lui-number"
  >
    100
  </span>
  <span
    class="lui-number"
  >
    100
  </span>
  <span
    class="lui-number"
  >
    1,000
  </span>
  <span
    class="lui-number"
  >
    1.000
  </span>
  <span
    class="lui-number"
  >
    1 000
  </span>
  <span
    class="lui-number"
  >
    1,000
  </span>
  <span
    class="lui-number"
  >
    1,000
  </span>
  <span
    class="lui-number"
  >
    1,000
  </span>
  <span
    class="lui-number"
  >
    1,000
  </span>
  <span
    class="lui-number"
  >
    10,000
  </span>
  <span
    class="lui-number"
  >
    10.000
  </span>
  <span
    class="lui-number"
  >
    10 000
  </span>
  <span
    class="lui-number"
  >
    10,000
  </span>
  <span
    class="lui-number"
  >
    10,000
  </span>
  <span
    class="lui-number"
  >
    10,000
  </span>
  <span
    class="lui-number"
  >
    10,000
  </span>
  <span
    class="lui-number"
  >
    100,000
  </span>
  <span
    class="lui-number"
  >
    100.000
  </span>
  <span
    class="lui-number"
  >
    100 000
  </span>
  <span
    class="lui-number"
  >
    100,000
  </span>
  <span
    class="lui-number"
  >
    1,00,000
  </span>
  <span
    class="lui-number"
  >
    100,000
  </span>
  <span
    class="lui-number"
  >
    100,000
  </span>
  <span
    class="lui-number"
  >
    1,000,000
  </span>
  <span
    class="lui-number"
  >
    1.000.000
  </span>
  <span
    class="lui-number"
  >
    1 000 000
  </span>
  <span
    class="lui-number"
  >
    1,000,000
  </span>
  <span
    class="lui-number"
  >
    10,00,000
  </span>
  <span
    class="lui-number"
  >
    1,000,000
  </span>
  <span
    class="lui-number"
  >
    1,000,000
  </span>
</div>
`;
